integration/raft/config_test.go

```diff
@@ -136,6 +136,89 @@ var _ = Describe("EndToEnd reconfiguration and onboarding", func() {
 			Expect(protoutil.BlockHeaderBytes(b1.Header)).To(Equal(protoutil.BlockHeaderBytes(b2.Header)))
 			Expect(protoutil.BlockHeaderBytes(b2.Header)).To(Equal(protoutil.BlockHeaderBytes(b3.Header)))
 		})
+
+		It("Can select leader by adding/removing", func() {
+			o1, o2, o3 := network.Orderer("orderer1"), network.Orderer("orderer2"), network.Orderer("orderer3")
+			orderers := []*nwo.Orderer{o1, o2, o3}
+			peer = network.Peer("Org1", "peer0")
+
+			By("Launching the orderers")
+			for _, o := range orderers {
+				runner := network.OrdererRunner(o)
+				ordererRunners = append(ordererRunners, runner)
+				process := ifrit.Invoke(runner)
+				ordererProcesses = append(ordererProcesses, process)
+			}
+
+			for _, ordererProc := range ordererProcesses {
+				Eventually(ordererProc.Ready()).Should(BeClosed())
+			}
+
+			By("Waiting for them to elect a leader")
+			findLeader(ordererRunners)
+
+			By("0. Creating a channel with o1,o2,o3")
+			network.CreateChannel("testchannel", o3, peer)
+
+			By("assert Block reception")
+			assertBlockReception(map[string]int{
+				"testchannel":   0,
+				"systemchannel": 1,
+			}, []*nwo.Orderer{o1, o2, o3}, peer, network)
+
+			fmt.Fprintf(GinkgoWriter, "0. Leader node = %d\n", findLeader(ordererRunners)) // 1,2,3
+
+			extendNetwork(network)
+			certificatesOfOrderers := refreshOrdererPEMs(network)
+
+			By("1. Removing o1")
```

Comment:
1) Are you sure we need to remove 2 consenters? Isn't removing a single one sufficient to reproduce the behavior?
2) When you remove a consenter - it might have been the leader, so the cluster needs to stabilize. 10 seconds might or might not be enough, so let's find another means to ensure the remaining consenters see the leader ?